src/Core/Analysis/AnalysisIssue.cs

```diff
@@ -118,4 +118,22 @@ public class AnalysisIssueLocation : IAnalysisIssueLocation
 
         public string LineHash { get; }
     }
+
+    public class TextRange : ITextRange
```

Comment:
Created a new class so it could be used in #2971

Reply:
Is this a VS text range or a Sonar text range? Or do  we use it for both?